beacon/app/app.go

```diff
@@ -196,6 +209,53 @@ func (app *BeaconApp) createRPCServer() error {
 	return nil
 }
 
+func (app *BeaconApp) startInitSync() error {
+	go func() {
+		for {
+			// nest for..loop to allow sleep at the end of the outter loop
+			for {
+				peerListResponse, err := app.p2pRPCClient.GetPeers(context.Background(), &empty.Empty{})
```

Comment:
this never loops more than once. Put this code into a function that returns if it was successful or not and then just use the outer for loop to retry.

Reply:
Good idea. Fixed.